src/additional_functions/helper.jl

```diff
@@ -37,11 +41,8 @@ function get_observed(rowind, data, semobserved; args = (), kwargs = NamedTuple(
     return observed_vec
 end
 
-function skipmissing_mean(mat)
-    means = Vector{Float64}(undef, size(mat, 2))
-    for i in 1:size(mat, 2)
-        @views means[i] = mean(skipmissing(mat[:, i]))
-    end
+function skipmissing_mean(mat::AbstractMatrix)
+    means = [mean(skipmissing(coldata)) for coldata in eachcol(mat)]
     return means
 end
```

Comment:
If you want we could even do
```suggestion
skipmissing_mean(mat::AbstractMatrix) = 
    [mean(skipmissing(coldata)) for coldata in eachcol(mat)]
```